src/components/workspace/side-content/AutograderCard.tsx

```diff
@@ -37,55 +13,39 @@ type AutograderCardProps = {
 
 class AutograderCard extends React.Component<AutograderCardProps, {}> {
   public render() {
-    const renderResult = (value: any) => {
-      /** A class which is the output of the show() function */
-      const ShapeDrawn = (window as any).ShapeDrawn;
-      if (typeof ShapeDrawn !== 'undefined' && value instanceof ShapeDrawn) {
-        return <CanvasOutput />;
-      } else {
-        return stringify(value);
-      }
-    };
+    let gradingStatus: string = '';
+
+    if (this.props.testcase.result) {
+      gradingStatus =
+        stringify(this.props.testcase.result) === this.props.testcase.answer
```

Comment:
remove ternary here also! The logic is not obvious

Reply:
Refactored